examples/record_loader.rb

```diff
@@ -1,18 +1,31 @@
 class RecordLoader < GraphQL::Batch::Loader
-  def initialize(model, column: model.primary_key, where: nil)
+  EMPTY_ARRAY = [].freeze
+
+  def initialize(model, column: model.primary_key, where: nil, unique: true)
     @model = model
-    @column = column.to_s
-    @column_type = model.type_for_attribute(@column)
+    @column = column
+    @column_type = model.type_for_attribute(@column.to_s)
     @where = where
+    @unique = unique
   end
 
   def load(key)
     super(@column_type.cast(key))
   end
 
   def perform(keys)
-    query(keys).each { |record| fulfill(record.public_send(@column), record) }
-    keys.each { |key| fulfill(key, nil) unless fulfilled?(key) }
+    if @unique
+      query(keys).each do |record|
+        fulfill(record.public_send(@column), record)
+      end
+      default = nil
+    else
+      query(keys).group_by(&@column).each do |key, result|
```

Comment:
* `group by` may not be a good choice without unique index (see: https://stackoverflow.com/questions/34115174/error-related-to-only-full-group-by-when-executing-a-query-in-mysql)

Reply:
This should be doing Array#group_by.  Perhaps `.to_a` should be used at the end of `query` to make this clear